From 93a6f87d9885f04b1c232abf0e215d74dc77d2d7 Mon Sep 17 00:00:00 2001
From: Pavan Kumar Ramayanam <pavan.ramayanam@amd.com>
Date: Fri, 27 Apr 2018 00:09:30 +0530
Subject: [PATCH 4590/5855] drm/amdkcl:[4.14] Force freesync_capable setting

Fixing freesync_capable property default setting
on kernels prior to 4.14

Acked-by: Alex Deucher <alexander.deucher@amd.com>
Acked-by: Junwei Zhang <Jerry.Zhang@amd.com>
Reviewed-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Roman Li <Roman.Li@amd.com>

Signed-off-by: Pavan Kumar Ramayanam <pavan.ramayanam@amd.com>
---
 drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
index 8fdb5f7..5a4ee3c 100644
--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
@@ -5066,6 +5066,9 @@ void amdgpu_dm_add_sink_to_freesync_module(
 			dm_con_state->freesync_capable = true;
 		}
 	}
+	drm_object_property_set_value(&connector->base,
+			              adev->mode_info.freesync_capable_property,
+				      dm_con_state->freesync_capable);
 }
 
 void amdgpu_dm_remove_sink_from_freesync_module(
@@ -5100,4 +5103,8 @@ void amdgpu_dm_remove_sink_from_freesync_module(
 	dm_con_state->user_enable.enable_for_gaming = false;
 	dm_con_state->user_enable.enable_for_static = false;
 	dm_con_state->user_enable.enable_for_video = false;
+
+	drm_object_property_set_value(&connector->base,
+				      adev->mode_info.freesync_capable_property,
+				      dm_con_state->freesync_capable);
 }
-- 
2.7.4

